rotkehlchen/usage_analytics.py

```diff
@@ -22,134 +24,123 @@
-    try:
-        response = requests.get(
-            'http://api.ipstack.com/check?access_key=affd920d6e1008a614900dbc31d52fa6',
-            timeout=LOCATION_DATA_QUERY_TIMEOUT,
-        )
-    except requests.exceptions.RequestException:
-        return None
-
-    if response.status_code != HTTPStatus.OK:
+        u = miniupnpc.UPnP()
+        u.discoverdelay = 200
+        u.discover()
+        u.selectigd()
+        user_ip = u.externalipaddress()
+    except Exception as e:  # pylint: disable=broad-except
+        # In my case I was getting a generic exception rised
+        # when failing to detect the ip
+        log.debug(f'Failed to get ip for analytics. {str(e)}')
         return None
 
     try:
-        json_ret = jsonloads_dict(response.text)
-    except JSONDecodeError:
-        return None
-
-    return GeolocationData(
-        country_code=json_ret.get('country_code', 'unknown'),
-        city=json_ret.get('city', 'unknown'),
-    )
-
-
-def retrieve_location_data() -> Optional[GeolocationData]:
-    # First try asking the no api key service
-    location_data = query_ipwhoisio()
-    if location_data:
-        return location_data
-
-    # Then the one with the most API requests per key
-    location_data = query_ipinfo()
-    if location_data:
-        return location_data
-
-    # finally the 3rd one as backup
-    location_data = query_ipstack()
-    # Can also be None if last query also fails
-    # TODO: Add more fallbacks
-    return location_data
+        with maxminddb.open_database(filename) as reader:
+            data = reader.get(user_ip)
+            location = data['country']['iso_code']
+            if location == 'None':
+                return None
+            return GeolocationData(country_code=location)
+    except maxminddb.errors.InvalidDatabaseError as e:
+        os.remove(filename)
+        log.debug(f'Failed to read database {str(e)}')
+    except ValueError as e:
+        log.debug(f'Wrong ip search {str(e)}')
+    return None
 
 
-def create_usage_analytics() -> Dict[str, Any]:
+def create_usage_analytics(data_dir: Path) -> Dict[str, Any]:
     analytics = {}
 
     analytics['system_os'] = platform.system()
     analytics['system_release'] = platform.release()
     analytics['system_version'] = platform.version()
     analytics['rotki_version'] = get_system_spec()['rotkehlchen']
 
-    location_data = retrieve_location_data()
+    location_data = retrieve_location_data(data_dir)
     if location_data is None:
-        location_data = GeolocationData('unknown', 'unknown')
+        location_data = GeolocationData('unknown')
 
     analytics['country'] = location_data.country_code
-    analytics['city'] = location_data.city
+    # To improvice privacy of the users we have changed the method of obtaining information
+    # about the country of the user and the city is not information we get anymore. We leave
+    # the field city for compatibility with the API.
+    analytics['city'] = 'unknown'
```

Comment:
This is a test. You just write a few words, not an entire paragraph :smile: 

```suggestion
    analytics['city'] = 'unknown'  # deprecated -- we no longer use it
```